solr/solr-ref-guide/modules/query-guide/pages/embedding-text.adoc

```diff
@@ -0,0 +1,269 @@
+The models currently supported are:
+
+[tabs#supported-models]
+======
+Hugging Face::
++
+====
+
+[source,json]
+----
+{
+  "class": "dev.langchain4j.model.huggingface.HuggingFaceEmbeddingModel",
+  "name": "<a-name-for-your-model>",
+  "params": {
+    "accessToken": "<your-huggingface-api-key>",
+    "modelId": "<a-huggingface-embedding-model>"
+  }
+}
+----
+====
+
+MistralAI::
++
+====
+[source,json]
+----
+{
+  "class": "dev.langchain4j.model.mistralai.MistralAiEmbeddingModel",
+  "name": "<a-name-for-your-model>",
+  "params": {
+    "baseUrl": "https://api.mistral.ai/v1",
+    "apiKey": "<your-mistralAI-api-key>",
+    "modelName": "<a-mistralAI-embedding-model>",
+    "timeout": 60,
+    "logRequests": true,
+    "logResponses": true,
+    "maxRetries": 5
+  }
+}
+----
+====
+
+OpenAI::
++
+====
+[source,json]
+----
+{
+  "class": "dev.langchain4j.model.openai.OpenAiEmbeddingModel",
+  "name": "<a-name-for-your-model>",
+  "params": {
+    "baseUrl": "https://api.openai.com/v1",
+    "apiKey": "<your-openAI-api-key>",
+    "modelName": "<a-openAI-embedding-model>",
+    "timeout": 60,
+    "logRequests": true,
+    "logResponses": true,
+    "maxRetries": 5
+  }
+}
+----
+====
+
+Cohere::
++
+====
+[source,json]
+----
+{
+  "class": "dev.langchain4j.model.cohere.CohereEmbeddingModel",
+  "name": "<a-name-for-your-model>",
+  "params": {
+    "baseUrl": "https://api.cohere.ai/v1/",
+    "apiKey": "<your-cohere-api-key>",
+    "modelName": "<a-cohere-embedding-model>",
+    "inputType": "search_document",
+    "timeout": 60,
+    "logRequests": true,
+    "logResponses": true
+  }
+}
+----
+====
+======
+
+=== Uploading a Model
+
+To upload the model in a `/path/myModel.json` file, please run:
+
+[source,bash]
+----
+curl -XPUT 'http://localhost:8983/solr/techproducts/schema/embedding-model-store' --data-binary "@/path/myModel.json" -H 'Content-type:application/json'
+----
+
+
+To view all models:
+
+[source,text]
+http://localhost:8983/solr/techproducts/schema/embedding-model-store
+
+To delete the `currentModel` model:
+
+[source,bash]
+----
+curl -XDELETE 'http://localhost:8983/solr/techproducts/schema/embedding-model-store/currentModel'
+----
+
+
+To view the model you just uploaded please open the following URL in a browser:
+
+[source,text]
+http://localhost:8983/solr/techproducts/schema/embedding-model-store
+
+.Example: /path/myModel.json
+[source,json]
+----
+{
+  "class": "dev.langchain4j.model.openai.OpenAiEmbeddingModel",
+  "name": "openai-1",
+  "params": {
+    "baseUrl": "https://api.openai.com/v1",
+    "apiKey": "apiKey-openAI",
+    "modelName": "text-embedding-3-small",
+    "timeout": 60,
+    "logRequests": true,
+    "logResponses": true,
+    "maxRetries": 5
+  }
+}
+
+----
+
+=== Running an embedding Query
+To run a query that embeds your query text, using a model you previously uploaded is simple:
+
+[source,text]
+?q={!text_to_vector model=a-model f=vector topK=10}hello world query
```

Comment:
Did you consider adding capabilities to KnnQParser instead of defining a new one?  Subclassing seems to lose the fact that we are doing KNN.  I suppose the issue is modularization, as KNN is in Solr-Core but what you are adding is elsewhere.  Nonetheless, some interfaces could be in solr-core.

Reply:
I considered that, but proceeded this way for simplicity, happy for others to evlove this later!